File: 111-120/Problem_120.R
r = c()

#(a???1)n + (a+1)n mod (a^2)

for(a in 3:1000)
{
  remainderSeries = c(2)
  n = 1
  r_max = 2
  found = FALSE
  while(!found)
  {
    new_r = (2*a*n)%%(a*a)
    if(new_r > r_max)
    {
      r_max = new_r
    }
    if(new_r %in% remainderSeries)
    {
      r = c(r,r_max)
      found = TRUE
    }
    else
    {
      remainderSeries = c(remainderSeries,new_r)
      n = n + 2
    }
  }
}
sum(r)

File: 111-120/Problem_120.py
#PROBLEM 120
from array import *

import time
start = time.time()

r = []

for a in range(3, 1001):
    RemainderSeries = [2]
    n = 1
    r_max = 2
    found = False
    while not found:
        new_r = (2 * a * n) % (a * a)
        if new_r > r_max:
            r_max = new_r
        if new_r in RemainderSeries:
            r.append(r_max)
            found = True
        else:
            RemainderSeries.append(new_r)
            n = n + 2
print(sum(r))
end = time.time()
print(end - start)
